Invalid memory address or nil pointer dereference in ETCD component

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area control-plane
/kind bug
/exp beginner

**What happened**:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1e37097]

goroutine 7610 [running]:
github.com/gardener/gardener/pkg/operation/botanist/component/etcd.(*etcd).Deploy(0xc0013c9c00, 0x292ddb0, 0xc001b05c80, 0x1, 0xc000bdc768)
        /home/vpnachev/go/src/github.com/gardener/gardener/pkg/operation/botanist/component/etcd/etcd.go:198 +0x337
github.com/gardener/gardener/pkg/utils/flow.Parallel.func1.1(0xc001eb9ee4, 0xc001f5e840, 0xc001b78510, 0x292ddb0, 0xc001b05c80)
        /home/vpnachev/go/src/github.com/gardener/gardener/pkg/utils/flow/taskfn.go:132 +0x66
created by github.com/gardener/gardener/pkg/utils/flow.Parallel.func1
        /home/vpnachev/go/src/github.com/gardener/gardener/pkg/utils/flow/taskfn.go:130 +0x105
```

**What you expected to happen**:
Optional fields to be validated before to be used here https://github.com/gardener/gardener/blob/29ffac5ed724458803b7748800e9de6f07c84f0a/pkg/operation/botanist/component/etcd/etcd.go#L198

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version: v1.34.0, 97e2e316891bfad3c7e381d7fea7d068f7453aa9
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/pkg/operation/botanist/component/etcd/etcd.go
+++ b/pkg/operation/botanist/component/etcd/etcd.go
@@ -195,7 +195,7 @@ func (e *etcd) Deploy(ctx context.Context) error {
 	}
 
 	stsName := e.etcd.Name
-	if foundEtcd && existingEtcd.Status.Etcd.Name != "" {
+	if foundEtcd && existingEtcd.Status.Etcd != nil && existingEtcd.Status.Etcd.Name != "" {
 		stsName = existingEtcd.Status.Etcd.Name
 	}
 

--- a/pkg/operation/botanist/component/etcd/etcd_test.go
+++ b/pkg/operation/botanist/component/etcd/etcd_test.go
@@ -635,6 +635,62 @@ var _ = Describe("Etcd", func() {
 				Expect(etcd.Deploy(ctx)).To(Succeed())
 			})
 
+			It("should not panic during deploy when etcd resource exists, but its status is not yet populated", func() {
+				oldTimeNow := TimeNow
+				defer func() { TimeNow = oldTimeNow }()
+				TimeNow = func() time.Time { return now }
+
+				var (
+					existingReplicas int32 = 245
+					retainReplicas         = true
+				)
+
+				etcd = New(c, log, testNamespace, testRole, class, retainReplicas, storageCapacity, &defragmentationSchedule)
+				setSecretsAndHVPAConfig()
+
+				gomock.InOrder(
+					c.EXPECT().Get(ctx, kutil.Key(testNamespace, etcdName), gomock.AssignableToTypeOf(&druidv1alpha1.Etcd{})).Return(apierrors.NewNotFound(schema.GroupResource{}, "")).DoAndReturn(func(ctx context.Context, _ client.ObjectKey, obj client.Object) error {
+						(&druidv1alpha1.Etcd{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      etcdName,
+								Namespace: testNamespace,
+							},
+							Spec: druidv1alpha1.EtcdSpec{
+								Replicas: int(existingReplicas),
+							},
+						}).DeepCopyInto(obj.(*druidv1alpha1.Etcd))
+						return nil
+					}),
+					c.EXPECT().Get(ctx, kutil.Key(testNamespace, etcdName), gomock.AssignableToTypeOf(&appsv1.StatefulSet{})).Return(apierrors.NewNotFound(schema.GroupResource{}, "")),
+
+					c.EXPECT().Get(ctx, kutil.Key(testNamespace, networkPolicyName), gomock.AssignableToTypeOf(&networkingv1.NetworkPolicy{})),
+					c.EXPECT().Patch(ctx, gomock.AssignableToTypeOf(&networkingv1.NetworkPolicy{}), gomock.Any()).Do(func(ctx context.Context, obj client.Object, _ client.Patch, _ ...client.PatchOption) {
+						Expect(obj).To(DeepEqual(networkPolicy))
+					}),
+					c.EXPECT().Get(ctx, kutil.Key(testNamespace, etcdName), gomock.AssignableToTypeOf(&druidv1alpha1.Etcd{})),
+					c.EXPECT().Patch(ctx, gomock.AssignableToTypeOf(&druidv1alpha1.Etcd{}), gomock.Any()).Do(func(ctx context.Context, obj *druidv1alpha1.Etcd, _ client.Patch, _ ...client.PatchOption) {
+						// ignore status when comparing
+						obj.Status = druidv1alpha1.EtcdStatus{}
+
+						Expect(obj).To(DeepEqual(etcdObjFor(
+							class,
+							int(existingReplicas),
+							nil,
+							"",
+							"",
+							nil,
+							nil,
+						)))
+					}),
+					c.EXPECT().Get(ctx, kutil.Key(testNamespace, hvpaName), gomock.AssignableToTypeOf(&hvpav1alpha1.Hvpa{})),
+					c.EXPECT().Patch(ctx, gomock.AssignableToTypeOf(&hvpav1alpha1.Hvpa{}), gomock.Any()).Do(func(ctx context.Context, obj client.Object, _ client.Patch, _ ...client.PatchOption) {
+						Expect(obj).To(DeepEqual(hvpaFor(class, existingReplicas, scaleDownUpdateMode)))
+					}),
+				)
+
+				Expect(etcd.Deploy(ctx)).To(Succeed())
+			})
+
 			It("should successfully deploy (normal etcd) and retain replicas (etcd found)", func() {
 				oldTimeNow := TimeNow
 				defer func() { TimeNow = oldTimeNow }()

